parler_tts/modeling_parler_tts.py

```diff
@@ -867,6 +958,36 @@ def forward(
                 )
 
         input_shape = inputs_embeds.size()[:-1]
+
+        if not self.config.rope_embeddings:
+            # embed positions
+            # TODO: As it is, the masked ids from the prompt will still count in the positions embeddings
+            # maybe should modify position embeddings
+            positions = self.embed_positions(inputs_embeds, past_key_values_length)
+            hidden_states = inputs_embeds + positions.to(inputs_embeds.device)
+        else:
+            hidden_states = inputs_embeds
+
+            if position_ids is None:
+                if attention_mask is not None:
```

Comment:
Ideally this logic would go in `prepare_inputs_for_generation` (e.g. if we did a formal Transformers integration). The reason we don't is that we do some pre-processing of the `prompt_attention_mask` and `attention_mask`, from which we compute the `position_ids`. This pre-processing currently only happens in the `.forward` pass, so it's cleaner to also compute the `position_ids` here as well

Reply:
So we create `position_ids` at every generation step ?